lib/template-dependencies.js

```diff
@@ -111,11 +111,42 @@ async function update(options, pathToElmJson) {
 
 // SPAWNING
 
+let elmJsonPromise;
+
 function spawnElmJsonAsync(args) {
-  return spawnAsync(elmJsonCLI, args, {
-    silent: true,
-    env: process.env
-  });
+  if (elmJsonPromise === undefined) {
+    elmJsonPromise = getExecutable({
+      name: 'elm-json',
+      version: '^0.2.8',
+      onProgress: (percentage) => {
+        const message = `Downloading elm-json... ${Math.round(
+          percentage * 100
+        )}%`;
+        process.stderr.write(
+          percentage >= 1
+            ? `${'Working...'.padEnd(message.length, ' ')}\r`
+            : `${message}\r`
```

Comment:
Why stderr and not stdout? Also, does this update the message on the same line? I am having a hard time seeing how this would work out.

I think it would be better if we just returned a string and `elm-tooling` handles the writing to the standard output.

That might be a bad idea of mine, The API looks a bit implementation-sensitive to me. 

FYI, I personally like [`ora`](https://www.npmjs.com/package/ora), or the underlying [`cli-spinner`](https://www.npmjs.com/package/cli-spinner). Just showing them to you in case you didn't know and you might like it.

Reply:
> Why stderr and not stdout?

`curl` prints its progress bar to stderr so I thought it was appropriate. I don’t care much about it – we can change it to stdout if you like.

> Also, does this update the message on the same line?

Yes. `\r` (Carriage Return) moves the cursor back to the start of the line.

> I think it would be better if we just returned a string and `elm-tooling` handles the writing to the standard output.

Why? A function that gets the percentage as a number and allows doing anything you want is the simplest and most flexible API I can think of. Why restrict people to only being able to print stuff to `stdout`? What if the want to print to `stderr` or show progress some completely different way?

> FYI, I personally like ora, or the underlying cli-spinner. Just showing them to you in case you didn't know and you might like it.

I just did the simplest possible progress output to get a discussion going. I was hoping you’d comment on how you prefer it to be shown. Let me know how you want it and I’ll change. Also feel free to push directly to this branch if you feel like fiddling with it yourself.

(Maybe for inspiration, in node-test-runner I added progress output not just for downloading elm-json, but for the whole run. For example, `Solving dependencies > Compiling > Starting tests` where each segment appears over time.)

Reply:
Yeah. Let's leave it for now, and I'll change it to whatever feels nicest later. I think you're right, that this is the best API for it, since we're not even necessarily in the context of a terminal.

I will change it so it doesn't print anything when running with `--report=json` though.

Thanks for the info on `\r`, good to know!